spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaConsumerFactory.java

```diff
@@ -445,68 +440,68 @@ private void checkInaccessible(Properties properties, Map<String, Object> modifi
 		}
 	}
 
-	@SuppressWarnings("resource")
 	protected Consumer<K, V> createKafkaConsumer(Map<String, Object> configProps) {
 		checkBootstrap(configProps);
 		Consumer<K, V> kafkaConsumer = createRawConsumer(configProps);
```

Comment:
Perhaps we could throw an exception here, if listeners are present and the created consumer is not an instance of `ExtendedKafkaConsumer`?

Reply:
On the other hand, Java doesn't support multiple inheritance (the user might already be extending some other class), so a hard failure would not be so good; perhaps log a warning that listeners must be invoked when closing a custom consumer.

Or, simply revert the code and have `ExtendedKafkaConsumer` wrap the proxy instead so that the consumer is not `Serializable`.

Reply:
I agree; makes sense; maybe also deprecate that method?

Reply:
I would let it live for now until we really fail for some peculiar use-case to learn from.
If we deprecate and remove it (I guess there was a reason behind that method), we may never learn the mentioned use-case 😉